### ISSUE TITLE
Set antialiasSamples=4 crashes on Mac

### ISSUE THREAD

**Issue by a-gagnon** [iTwin/itwinjs-core#6033]
**Describe the bug**
This is an issue that @ezmobivietnam is experiencing with our civil views using his Macbook Pro 2019. Whenever the `antialiasSamples` is set, the WebGL context is lost and nothing works anymore.

**To Reproduce**
Please refer to the following sandbox:
https://www.itwinjs.org/sandboxes/gagnon-a/webgl_context_macos_binh

Loading the sandbox should be sufficient to reproduce the issue. The crux of the problem is the following lines:
```ts
  const onViewportRef = (vp: ScreenViewport) => {
    vp.antialiasSamples = 4;
    console.log('Adjusted to 4 antialias samples');
  }
```

**Expected behavior**
Changing a setting that is not supported by the user's browser/WebGL capabilities should not crash the app. Ideally, it should just ignore it.

**Screenshots**
![Screenshot 2023-09-21 at 3 17 21 PM](https://github.com/iTwin/itwinjs-core/assets/3418768/5c475bf0-f0a1-4ad6-840e-35de37333e41)
![Screenshot 2023-09-21 at 2 33 30 PM](https://github.com/iTwin/itwinjs-core/assets/3418768/29d6e3ff-8645-4dcc-8087-8a9804bcb5c0)

**Desktop (please complete the applicable information):**

Macbook Pro 2019
- Processor: 2.6 GHz 6-Core Intel Core i7
- Graphics: 
     AMD Radeon Pro 5300M 4 GB
     Intel UHD Graphics 630 1536 MB
- Memory: 16 GB 2667 MHz DDR4
- macOS: Ventura 13.5.2 (22G91)
- Getting the same results with Chrome, Edge and Firefox
- iTwin.js Version: 3.7.7 (sandbox), 4.x (deployed pineapple)

**Comment by pmconne** [comment]
Please send to the [compatibility checker](https://connect-imodeljscompatibility.bentley.com/) and report back with the number of antialias samples reported and all of the system information.
![image](https://github.com/iTwin/itwinjs-core/assets/22944042/7f50ab45-d3d8-4a66-94ea-4458e412e828)
![image](https://github.com/iTwin/itwinjs-core/assets/22944042/cede0f54-fe29-465a-a6eb-21f0515f55b3)

Then please find out how to configure the MacBook/browser to use the GPU. It is currently using software rendering, which is inappropriate for any 3D application.

**Comment by pmconne** [comment]
> here is the screenshot of the compatibility checker.

Did you change your browser to use your GPU? Because @a-gagnon's screenshot shows software rendering and "major performance caveat", while your screenshot shows hardware rendering an no such caveat.

**Comment by a-gagnon** [comment]
> > here is the screenshot of the compatibility checker.
> 
> Did you change your browser to use your GPU? Because @a-gagnon's screenshot shows software rendering and "major performance caveat", while your screenshot shows hardware rendering an no such caveat.

I took the screen caps from an earlier conversion to create this bug report, perhaps some of it is out of context (?)
@ezmobivietnam were you getting the same errors using the sandbox and the civil model from infrastructure cloud?

**Comment by ezmobivietnam** [comment]
> > here is the screenshot of the compatibility checker.
> 
> Did you change your browser to use your GPU? Because @a-gagnon's screenshot shows software rendering and "major performance caveat", while your screenshot shows hardware rendering an no such caveat.

No, I change nothing. I use the default browser settings. Regarding the performance caveat, it happens randomly. I just got it two times.

**Comment by pmconne** [comment]
@markschlosseratbentley @iTwin/itwinjs-core-display

**Comment by markschlosseratbentley** [comment]
@a-gagnon @ezmobivietnam 

Are you saying that your browser randomly goes from using a GPU to not using a GPU every time you run the compatibility checker?
If so, that sounds like a configuration problem.

**Comment by pmconne** [comment]
@ezmobivietnam's latest repro steps make no mention of antialiasing.

Somebody please put together a single coherent set of repro steps and put them in the issue description.

**Comment by ezmobivietnam** [comment]
@pmconne , I have no idea how the antialiasSamples is set/managed in Pineapple. But I did some tests with [Alex's sandbox ](https://www.itwinjs.org/sandboxes/gagnon-a/webgl_context_macos_binh) then had the result as below:

- Case 1: set  `vp.antialiasSamples` = to 4 then got the error:
![Screenshot 2023-10-01 at 2 07 47 PM](https://github.com/iTwin/itwinjs-core/assets/7710962/51d8c63c-5f33-46fb-a601-536647c2fb77)
- Case 2: don't set `vp.antialiasSamples`, the sample works well :
![Screenshot 2023-10-01 at 2 09 57 PM](https://github.com/iTwin/itwinjs-core/assets/7710962/0f880e0b-9995-466a-8215-574d92e1aa7f)

**Comment by simnorm** [comment]
@pmconne @markschlosseratbentley   Any updates on this? Do you need something else to complete this investigation?
He gets the error consistently using the Sandbox Alex provided.

**Comment by pmconne** [comment]
I fail to repro in Chrome and Firefox on Windows 11 and Ubuntu 22.04. This includes setting `vp.antialiasSamples` to a definitely unsupported value like 192.
I have no access to a macbook to figure out why your WebGL implementation is failing when you set the number of antialias samples to a value it claims it supports.
@markschlosseratbentley, do you?

**Comment by ezmobivietnam** [comment]
Hi @markschlosseratbentley ,

I have changed Chrome settings to enable/disable hardware acceleration, and tested the sandbox that Alex provided then got the result below:
1. Use hardware acceleration when available = ON (default setting): the issue consistently happens
2. Use hardware acceleration when available = OFF: the issue disappears. 

Thanks so much for your help!
Binh
